### PR TITLE
[EFR32]Feature wifinetworkdiagnostics implmentation for RS9116

### PR DESCRIPTION
**Pull request by chirag-silabs**
#### Problem
Missing implementation for wifinetworkdiagnostics cluster for RS9116

#### Change overview
Added implementation for wifinetworkdiagnostics for EFR32+RS9116 combo

#### Testing
Tested using EFR32+RS9116. 
TC-DGWIFI-2.1 tested

### CODE REVIEW

**Comment by chirag-silabs** [comment]
This depends upon #20106

**Review by rcasallas-silabs — APPROVED** [review]

**Review by mrjerryjohns — APPROVED** [review]

File: src/platform/EFR32/DiagnosticDataProviderImpl.cpp [hunk 1]
```diff
@@ -274,5 +341,143 @@ void DiagnosticDataProviderImpl::ReleaseNetworkInterfaces(NetworkInterface * net
...
 }
 
+#if SL_WIFI
+CHIP_ERROR DiagnosticDataProviderImpl::GetWiFiBssId(ByteSpan & BssId)
+{
+    wfx_wifi_scan_result_t ap;
+    int32_t err = wfx_get_ap_info(&ap);
+    if (err == 0)
+    {
+        BssId = ByteSpan(ap.bssid, 6);
```
Comment (bzbarsky-apple) [inline src/platform/EFR32/DiagnosticDataProviderImpl.cpp:351]: This is handing out a pointer to an on-stack data structure, no?  How is that supposed to work?

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/platform/EFR32/DiagnosticDataProviderImpl.cpp (hunk 1 above)
Comment (chirag-silabs) [reply src/platform/EFR32/DiagnosticDataProviderImpl.cpp:351]: Sorry, didn't understand what you meant. It gets the details via the platform call and update the value ap and returns the required value accordingly

Comment (bzbarsky-apple) [reply src/platform/EFR32/DiagnosticDataProviderImpl.cpp:351]: It's handing the caller the `ap.bssid` value.  That's a pointer, right?  What is that pointer pointing to?

Comment (chirag-silabs) [reply src/platform/EFR32/DiagnosticDataProviderImpl.cpp]: Got it, resolved this in the latest commit

File: src/platform/EFR32/DiagnosticDataProviderImpl.cpp [hunk 2]
```diff
@@ -341,5 +341,145 @@ void DiagnosticDataProviderImpl::ReleaseNetworkInterfaces(NetworkInterface * net
...
 
+#if SL_WIFI
+CHIP_ERROR DiagnosticDataProviderImpl::GetWiFiBssId(ByteSpan & BssId)
+{
+    wfx_wifi_scan_result_t ap;
+    int32_t err = wfx_get_ap_info(&ap);
+    static uint8_t bssid[6];
+    if (err == 0)
+    {
+        memcpy(bssid,ap.bssid,6);
```
Comment (bzbarsky-apple) [inline src/platform/EFR32/DiagnosticDataProviderImpl.cpp:352]: Going to need a restyle here...

**Review by bzbarsky-apple — APPROVED** [review]

File: src/platform/EFR32/DiagnosticDataProviderImpl.cpp (hunk 2 above)
Comment (chirag-silabs) [reply src/platform/EFR32/DiagnosticDataProviderImpl.cpp]: Sorry my bad. Done on the latest commit

**Pull request merged**